lib/src/logger.dart

```diff
@@ -139,7 +142,24 @@ class Logger {
       throw UnsupportedError(
           'Cannot set the level to `null` on a logger with no parent.');
     }
+    final isLevelChanged = _level != value;
     _level = value;
+    if (isLevelChanged) {
+      _levelChangedController?.add(value);
+    }
+  }
+
+  /// Returns a stream of level values set to this [Logger].
+  ///
+  /// You can listen for set levels using the standard stream APIs,
+  /// for instance:
+  ///
+  /// ```dart
+  /// logger.onLevelChanged.listen((level) { ... });
+  /// ```
```

Comment:
Should we add a note here that you should never set the level from within this callback? As doing so would create an infinite loop, but I could imagine somebody trying to use this to always force the level to a certain thing, and then two fighting implementations would toggle it back and forth forever lol.

Reply:
Good suggestion! I've just added a row to the API doc. I added also a test. Since a state error is thrown if the level is set from inside the callback, there won't be a loop forever.

Reply:
It is good suggestion! I've just added a row to the API doc. I added also a test. Since a state error is thrown if the level is set from inside the callback, there won't be a loop forever.